Common/TableProducer/centralityTable_new.cxx

```diff
@@ -0,0 +1,149 @@
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/RunningWorkflowInfo.h"
+#include "Common/DataModel/Multiplicity.h"
+#include "Common/DataModel/Centrality_new.h"
+#include <CCDB/BasicCCDBManager.h>
+#include <TH1F.h>
+#include <TF1.h>
+
+using namespace o2;
+using namespace o2::framework;
+
+struct CentralityTableTask {
+  Produces<aod::CentV0Ms> centVOM;
+  Produces<aod::CentSPDs> centSPD;
+  Produces<aod::CentCL0s> centCL1;
+  Produces<aod::CentCL1s> centCL2;
+  Service<o2::ccdb::BasicCCDBManager> ccdb;
+
+  Configurable<int> estV0M{"estV0M", -1, {"Produces centrality percentiles using V0 multiplicity. -1: auto, 0: don't, 1: yes. Default: auto (-1)"}};
+  Configurable<int> estSPD{"estSPD", -1, {"Produces centrality percentiles using SPD tracklets multiplicity. -1: auto, 0: don't, 1: yes. Default: auto (-1)"}};
+  Configurable<int> estCL0{"estCL0", -1, {"Produces centrality percentiles using CL0 multiplicity. -1: auto, 0: don't, 1: yes. Default: auto (-1)"}};
+  Configurable<int> estCL1{"estCL1", -1, {"Produces centrality percentiles using CL1 multiplicity. -1: auto, 0: don't, 1: yes. Default: auto (-1)"}};
+
+  int mRunNumber;
+  bool mV0MCallibrationStored;
+  TH1* mhVtxAmpCorrV0A;
+  TH1* mhVtxAmpCorrV0C;
+  TH1* mhMultSelCalibV0M;
+
+  void init(InitContext& context)
+  {
+    /* Checking the tables which are requested in the workflow and enabling them */
+    auto& workflows = context.services().get<RunningWorkflowInfo const>();
+    for (DeviceSpec device : workflows.devices) {
+      for (auto input : device.inputs) {
+        auto enable = [&input](const std::string detector, Configurable<int>& flag) {
+          const std::string table = "Cent" + detector + "s";
+          if (input.matcher.binding == table) {
+            if (flag < 0) {
+              flag.value = 1;
+              LOGF(info, "Auto-enabling table: %s", table.c_str());
+            } else if (flag > 0) {
+              flag.value = 1;
+              LOGF(info, "Table %s already enabled", table.c_str());
+            } else {
+              LOGF(info, "Table %s disabled", table.c_str());
+            }
+          }
+        };
+        enable("V0M", estV0M);
+        enable("SPD", estSPD);
+        enable("CL0", estCL0);
+        enable("CL1", estCL1);
+      }
+    }
+    ccdb->setURL("http://alice-ccdb.cern.ch");
+    ccdb->setCaching(true);
+    ccdb->setLocalObjectValidityChecking();
+    mRunNumber = 0;
+    mV0MCallibrationStored = false;
+    mhVtxAmpCorrV0A = nullptr;
+    mhVtxAmpCorrV0C = nullptr;
+    mhMultSelCalibV0M = nullptr;
+  }
+
+  void process(soa::Join<aod::Collisions, aod::Mults>::iterator const& collision, aod::BCsWithTimestamps const&, aod::Tracks const& tracks)
+  {
+    /* check the previous run number */
+    auto bc = collision.bc_as<aod::BCsWithTimestamps>();
+    if (bc.runNumber() != mRunNumber) {
```

Comment:
This is for caching, right? In the future the CCDBConfigurable will do this automatically.

Reply:
Good!
For the time being if the run number does not change the CCDB is only queried once